internal/sponge/item/item.go

```diff
@@ -0,0 +1,282 @@
+package item
+
+import (
+	"errors"
+	"time"
+
+	"github.com/ardanlabs/kit/db"
+	"github.com/ardanlabs/kit/db/mongo"
+	"github.com/ardanlabs/kit/log"
+
+	gc "github.com/patrickmn/go-cache"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// Contains the name of Mongo collections.
+const (
+	Collection        = "coral_items"
+	CollectionHistory = "coral_items_history"
+)
+
+// Set of error variables.
+var (
+	ErrNotFound = errors.New("Item Not found")
+)
+
+// =============================================================================
+
+// c contans a cache of set values. The cache will maintain items for one
+// second and then marked as expired. This is a very small cache so the
+// gc time will be every hour.
+
+const (
+	expiration = time.Second
+	cleanup    = time.Hour
+)
+
+var cache = gc.New(expiration, cleanup)
+
+// =============================================================================
+
+// create an item out of its type, version and data or die trying
+//  if the item already exists, adopt it's Id but do not presume to update
+func Create(context interface{}, db *db.DB, t string, v int, d map[string]interface{}) (Item, error) {
+
+	i := Item{}
```

Comment:
Smell Test:

Prefer to use ```var i Item```

i is generally a counter.  "Tough variable" to understand.

It's cleaner to build up different variables throughout the func then set them to an item at the end.